internal/controllers/inbound_orders/create.go

```diff
@@ -50,7 +50,12 @@ func (ctl *InboundOrdersController) Create(w http.ResponseWriter, r *http.Reques
 	inb, err := ctl.SV.Create(inboundOrders)
 	if err != nil {
 		// Check if the error is a MySQL duplicate entry error
-		if mysqlErr, ok := err.(*mysql.MySQLError); ok && mysqlErr.Number == mysqlerr.CodeCannotAddOrUpdateChildRow || mysqlErr.Number == mysqlerr.CodeDuplicateEntry {
+		if mysqlErr, ok := err.(*mysql.MySQLError); ok && mysqlErr.Number == mysqlerr.CodeDuplicateEntry {
+			response.Error(w, http.StatusConflict, err.Error())
+			return
+		}
+
+		if mysqlErr, ok := err.(*mysql.MySQLError); ok && mysqlErr.Number == mysqlerr.CodeCannotAddOrUpdateChildRow {
```

Comment:
acho que o erro que você comentou era por causa disso, você precisa por parenteses pra fazer os condicionais juntos:
```suggestion
		if mysqlErr, ok := err.(*mysql.MySQLError); ok &&
			(mysqlErr.Number == mysqlerr.CodeDuplicateEntry ||
				mysqlErr.Number == mysqlerr.CodeCannotAddOrUpdateChildRow) {
			response.Error(w, http.StatusConflict, err.Error())
			return
		}
```

Reply:
perfeito, vou commitar sua sugestão, obgdao petra :)